01DetectingFakeNews/01DetectingFakeNews.ipynb

```diff
@@ -0,0 +1,280 @@
+        "outputId": "aabb2b0f-0d71-4b92-86ab-0c2893416391"
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "PassiveAggressiveClassifier(max_iter=50)"
+            ]
+          },
+          "execution_count": 7,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "from sklearn.linear_model import PassiveAggressiveClassifier\n",
+        "pac=PassiveAggressiveClassifier(max_iter=50)\n",
+        "pac.fit(v_train,y_train)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Uco2a7Kafs-8"
+      },
+      "source": [
+        "## Predict the test set results"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 8,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "-vyXJTXHfxd1",
+        "outputId": "dcf4a115-cc73-44ae-ad77-a8a4c5b79d25"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Accuracy: 92.27%\n"
+          ]
+        }
+      ],
+      "source": [
+        "from sklearn.metrics import accuracy_score\n",
+        "y_pred=pac.predict(v_test)\n",
+        "score=accuracy_score(y_test,y_pred)\n",
+        "print(f'Accuracy: {round(score*100,2)}%')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "iCe3-vTbf_Lr"
+      },
+      "source": [
+        "## Confusion Matrix"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 9,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "TXOinqWegAaP",
+        "outputId": "a3e15571-b163-4d5c-da2b-1854e924b634"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "[[584  54]\n",
+            " [ 44 585]]\n"
+          ]
+        }
+      ],
+      "source": [
+        "from sklearn.metrics import confusion_matrix\n",
+        "cm = confusion_matrix(y_test, y_pred)\n",
+        "print(cm)"
+      ]
+    }
+  ],
+  "metadata": {
+    "colab": {
+      "collapsed_sections": [],
+      "provenance": []
+    },
+    "kernelspec": {
+      "display_name": "Python 3.9.12 ('base')",
+      "language": "python",
+      "name": "python3"
+    },
+    "language_info": {
+      "codemirror_mode": {
+        "name": "ipython",
+        "version": 3
+      },
+      "file_extension": ".py",
+      "mimetype": "text/x-python",
+      "name": "python",
+      "nbconvert_exporter": "python",
+      "pygments_lexer": "ipython3",
+      "version": "3.9.12"
+    },
+    "vscode": {
+      "interpreter": {
+        "hash": "ffb21c7f50ccbdd9cabeaacb722f478577479649672a627763d22c6e44daa1fc"
+      }
+    }
+  },
+  "nbformat": 4,
+  "nbformat_minor": 0
+}
```

Comment:
nice job! as an exercise I would like you to use https://scikit-learn.org/stable/modules/grid_search.html to select the best classification model (it is a more generic solution)
Leave the code as it is and continue with this grid search (so don't alter the existing code, but continue the notebook with a new lines)

Reply:
you can also try other automatic AI model selection algorithms, like AutoSklearn, Tpot and others